src/window_identifier/gtk4.rs

```diff
@@ -26,35 +21,17 @@ impl Gtk4WindowIdentifier {
         let surface = native.surface();
         match surface.display().backend() {
             Backend::Wayland => {
-                let top_level = surface
+                use gdk4wayland::prelude::WaylandSurfaceExtManual;
```

Comment:
You need to modify the drop implementation

Reply:
Also commit: 
```
WindowIdentifier/gtk4: Use wayland crate for exporting the surface

As we already need wayland for the activation token support,see next commit.
This will also avoid us having custom logic that is needed for keeping an internal ref counted token
because gtk4 doesn't support calling export_handle on an already exported surface.
```

Reply:
Also this function no longer needs to be async, yay!